### ISSUE TITLE
Occasional volume change on Apple TV

### ISSUE THREAD

**Issue by chucksense** [ejurgensen/forked-daapd#368]
Setup:
I use forked-daapd (built from source here) on a headless Ubuntu server and I'm generally playing simultaneously to two AirPlay destinations: an airport express and a latest-gen appletv. 

I control using the iOS Remote.app

I've noticed that sometimes after changing volumes, about 90 seconds later the volume resets to what seems like an arbitrary (and generally higher than I like) value. 

Let me know if there is any other detail I can provide. And thank you for being so awesomely responsive!

**Comment by ejurgensen** [comment]
Yes, forked-daapd will send a volume command to ATV's after about 90 seconds. There was an issue a couple months ago that required this as a work-around. Of course, it's supposed to be the correct volume. I will check if the work-around is still required, and if so, try to figure out why the correct volume isn't sent.

**Comment by anjo03** [comment]
Hey there. This bug is still present. During first playback in the morning, for an example, it seems like the 90 seconds is correct. But after that it can be between 5 and 90 seconds, then the volume will be set to something random (it seems like). But suddenly after like 6-15 changes, and I cannot determine the reason, the volume will stay static. Checking with "mpc volume", there are no changes to the players volume setting, so it must be something internally between forked-daapd and Apple TV...

**Comment by ejurgensen** [comment]
forked-daapd sends the volume every 60 secs to ATV's to keep them alive (there was another bug where they would disconnect). Should be ok if the right volume is sent, but apparently not always so. Could you try if the fix referenced above works? (in the fixvolume1 branch, so 'git checkout fixvolume1').

**Comment by anjo03** [comment]
Hi again.

The bug with the volume change seems to be fixed… or actually I do not know, because I experienced a few crashes:

 

Jun 26 19:35:11 server01.internal.naj.dk systemd[1]: Started Process Core Dump (PID 26110/UID 0).

Jun 26 19:35:11 server01.internal.naj.dk systemd[1]: forked-daapd.service: Main process exited, code=killed, status=11/SEGV

Jun 26 19:35:11 server01.internal.naj.dk systemd[1]: forked-daapd.service: Unit entered failed state.

Jun 26 19:35:11 server01.internal.naj.dk systemd[1]: forked-daapd.service: Failed with result 'signal'.

 

I just tried to run the program in the foreground manually, and have not experienced any crash yet.

And the volume is not changing by itself anymore ☺ 

 

/Anders
